### PR TITLE
[PRODUCT-1176] E2E tests for coherent simulation

### PR DESCRIPTION
**Pull request by ryanbarker**
Added tests for each scenario in the ticket. (Also includes some updates from https://github.com/qctrl/python/pull/84).

### CODE REVIEW

File: qctrl/services/coherent_simulations.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from qctrlapiclient import resourceobject
+from .base import BaseService
+
+# @pylint: disable=too-few-public-methods
+class CoherentSimulations(BaseService):
+    """ A service for calculating noise free metrics for a system.
```
Comment (stabacco) [inline qctrl/services/coherent_simulations.py]: wrong docstring?

File: tests/test_quasi_static_functions.py [hunk 2]
```diff
@@ -191,7 +191,7 @@ def test_quasi_static_function_evaluation_additive_noise(qctrl, system):
     )
 
     quasi_static_function = qctrl.services.quasi_static_functions.calculate(quasi_static_function)
-    assert quasi_static_function.sampled_points
+    assert quasi_static_function.sampled_points is not None
```
Comment (stabacco) [inline tests/test_quasi_static_functions.py]: could we maybe check that it's a list and that its length is what we're expecting? Same goes for the other lists.

File: qctrl/services/coherent_simulations.py (hunk 1 above)
Comment (ryanbarker) [reply qctrl/services/coherent_simulations.py]: I thought I had remembered to change it - guess not. I'll update soon.

File: tests/test_quasi_static_functions.py (hunk 2 above)
Comment (ryanbarker) [reply tests/test_quasi_static_functions.py]: Its going to be a numpy array now with the updates to api-client. Good idea checking the expected length.

**Pull request merged**